packages/core/data-transfer/src/engine/index.ts

```diff
@@ -240,6 +270,28 @@ class TransferEngine<
     }
   }
 
+  shouldSkipStage(stage: TransferStage) {
+    const { exclude, only } = this.options;
+
+    // everything is included by default unless 'only' has been set
+    let included = isEmpty(only);
+    if (only?.length > 0) {
+      included = only.some((transferGroup) => {
+        return TransferGroupPresets[transferGroup][stage];
+      });
+    }
+
+    if (exclude?.length > 0) {
+      if (included) {
+        included = !exclude.some((transferGroup) => {
+          return TransferGroupPresets[transferGroup][stage];
+        });
+      }
+    }
+
+    return !included;
+  }
```

Comment:
Should we check somewhere (maybe even at command level) that only and exclude don't have the same options? It might improve the DX somehow

Reply:
Good idea, added a check!

One thing to note though is that I've intentionally allowed both options to be used together, because in the future when we add categories that run a stage but filter it (such as assets being removed from entities) someone may want to run, for example, "--only media --exclude files" to get just their media entities+links without the files.